client/transport/streamable_http.go

```diff
@@ -307,6 +299,66 @@ func (c *StreamableHTTP) SendRequest(
 	}
 }
 
+func (c *StreamableHTTP) sendHTTP(
+	ctx context.Context,
+	method string,
+	body io.Reader,
+	acceptType string,
+) (resp *http.Response, err error) {
+
+	// Create HTTP request
+	req, err := http.NewRequestWithContext(ctx, method, c.serverURL.String(), body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set headers
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", acceptType)
+	sessionID := c.sessionID.Load().(string)
+	if sessionID != "" {
+		req.Header.Set(headerKeySessionID, sessionID)
+	}
+	for k, v := range c.headers {
+		req.Header.Set(k, v)
+	}
+
+	// Add OAuth authorization if configured
+	if c.oauthHandler != nil {
+		authHeader, err := c.oauthHandler.GetAuthorizationHeader(ctx)
+		if err != nil {
+			// If we get an authorization error, return a specific error that can be handled by the client
+			if err.Error() == "no valid token available, authorization required" {
+				return nil, &OAuthAuthorizationRequiredError{
+					Handler: c.oauthHandler,
+				}
+			}
+			return nil, fmt.Errorf("failed to get authorization header: %w", err)
+		}
+		req.Header.Set("Authorization", authHeader)
+	}
+
+	if c.headerFunc != nil {
+		for k, v := range c.headerFunc(ctx) {
+			req.Header.Set(k, v)
+		}
+	}
+
+	// Send request
+	resp, err = c.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send request: %w", err)
+	}
+
+	// universal handling for session terminated
+	if resp.StatusCode == http.StatusNotFound {
+		c.sessionID.CompareAndSwap(sessionID, "")
+		return nil, errSessionTerminated
+	}
+
+	return resp, nil
+}
+
 // handleSSEResponse processes an SSE stream for a specific request.
 // It returns the final result for the request once received, or an error.
 func (c *StreamableHTTP) handleSSEResponse(ctx context.Context, reader io.ReadCloser) (*JSONRPCResponse, error) {
```

Comment:
When `readSSE#ReadString` occur error, `reader` will be closed and won't receive notification anymore, but `handleSSEResponse` will wait `ctx.Done()`. Now, client will ignore all the notification, am I right?
![image](https://github.com/user-attachments/assets/1df4d311-290f-425e-8e1a-f0a5d649f08c)

![image](https://github.com/user-attachments/assets/0d8d4003-cce8-41b3-968d-3e0cec66e67f)

Reply:
<img width="759" alt="image" src="https://github.com/user-attachments/assets/8c32b38a-4483-4462-a835-9af6d4b350bd" />
when readSSE is done, the responseChan  will be closed, and goes 
<img width="553" alt="image" src="https://github.com/user-attachments/assets/717e160c-961e-4543-aff9-7b2ba0644cf5" />